File: Deepseek-Coder-33b-Instruct/stage2/loop_construct_collapse_clause.cpp
#include <iostream>
#include <cmath>
#include <cstdlib>
#include <openacc.h>

#define N 100

int test1(){
    int err = 0;
    int a[N][N];
    srand(1234);

    // Initialize the array
    for(int i=0; i<N; i++)
        for(int j=0; j<N; j++)
            a[i][j] = rand() % 100;

    // Use OpenACC to collapse the two loops
    #pragma acc parallel loop collapse(2)
    for(int i=0; i<N; i++)
        for(int j=0; j<N; j++)
            a[i][j] = a[i][j] * 2;

    // Check the result
    for(int i=0; i<N; i++)
        for(int j=0; j<N; j++)
            if(a[i][j] != rand() % 100 * 2)
                err = 1;

    return err;
}

int main(){
    int failcode = 0;
    int failed;

    failed = 0;
    for (int x = 0; x < 1; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }

    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/loop_construct_collapse_clause.f90
#ifndef T1
!T1:loop construct collapse clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER, PARAMETER :: N = 100
        REAL, DIMENSION(N,N) :: a, b, c
        INTEGER :: i, j, errors = 0

        ! Initialize arrays
        DO i = 1, N
          DO j = 1, N
            a(i,j) = i*j
            b(i,j) = i+j
          END DO
        END DO

        ! Perform operations on 2D array
        !$acc parallel loop collapse(2)
        DO i = 1, N
          DO j = 1, N
            c(i,j) = a(i,j) + b(i,j)
          END DO
        END DO

        ! Check results
        DO i = 1, N
          DO j = 1, N
            IF (c(i,j) .ne. a(i,j) + b(i,j)) THEN
              errors = errors + 1
            END IF
          END DO
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM